### PR TITLE
Feature/refacto comment

### PR DESCRIPTION
**Pull request by hermannleboss**
## Description

refacto comment

---

---

### CODE REVIEW

File: src/components/Posts/Comments/DisplayComment.vue [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    .finally(() => (isEditing.value = false))
+}
+
+async function deleteComment(commentId) {
+  await commentStore
+    .deleteComment(props.collectionName, props.documentId, commentId)
+    .then(() => $q.notify({ type: 'positive', message: 'Comment successfully deleted' }))
+    .catch((error) => errorStore.throwError(error, 'Failed to delete comment'))
+
+  // TODO Set all child comments parentId into the current comment Parent Id)
```
Comment (arnonrdp) [inline src/components/Posts/Comments/DisplayComment.vue:151]: Are you still going to do this or is it already done?

File: src/components/Posts/Comments/DisplayComment.vue [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+const replyCounter = (id) => {
+  return commentStore.getComments ? commentStore.getComments.filter((comment) => comment.parentId === id).length : 0
+}
+
+function handleKeydown(event) {
+  if (event.key === 'Escape' || event.key === 'Esc') {
+    // expanded.value = false
+    // commentId.value = ''
+    // expandedReply.value = false
+    // inputField.value.blur()
```
Comment (arnonrdp) [inline src/components/Posts/Comments/DisplayComment.vue:179]: Is it needed?

File: src/components/Posts/Comments/DisplayComment.vue [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+          <q-tooltip anchor="bottom middle" self="center middle">Reply</q-tooltip>
+        </q-btn>
+      </div>
+      <DisplayComment
+        v-for="item of commentStore.getCommentChildren(comment.id)"
+        :document-id="documentId"
+        :collection-name="collectionName"
+        :key="item.id"
+        :comment="item"
+      ></DisplayComment>
```
Comment (arnonrdp) [inline src/components/Posts/Comments/DisplayComment.vue:101]: Interesting solution.

File: src/pages/PromptPage.vue [hunk 4]
```diff
@@ -67,13 +68,14 @@ const prompt = computed(() => {
...
 
-  if (!prompt.value) {
-    router.push('/404')
-    return
-  }
+  setTimeout(() => {
+    if (!prompt.value) {
+      router.push('/404')
+    }
+  }, 5000)
```
Comment (arnonrdp) [inline src/pages/PromptPage.vue:78]: I don't think this is necessary. You can just place the `await` outside the condition.

**Review by arnonrdp — CHANGES_REQUESTED** [review]
Here are some points I noticed.

I've also been wondering about the `deleteComment` function.
Currently on the Backend we are only editing the comments for "Comment Deleted".

What if we created the following condition:
- If the comment has replies, edit it.
- Else, delete it.

In this case, it's just a suggestion.
What do you think?

**Review by arnonrdp — CHANGES_REQUESTED** [review]
Here are some points I noticed.

I've also been wondering about the `deleteComment` function.
Currently on the Backend we are only editing the comments for "Comment Deleted".

What if we created the following condition:
- If the comment has replies, edit it.
- Else, delete it.

In this case, it's just a suggestion.
What do you think?

File: src/components/Posts/Comments/DisplayComment.vue (hunk 1 above)
Comment (hermannleboss) [reply src/components/Posts/Comments/DisplayComment.vue:151]: It's not done, now we need to discuss how the comment will work. Then we can decide if we will do it or not

File: src/pages/PromptPage.vue (hunk 4 above)
Comment (hermannleboss) [reply src/pages/PromptPage.vue:78]: I think this `await new Promise((resolve) => setTimeout(resolve, 5000)) ` is blocking 
And this is async 
```js
setTimeout(() => {
    if (!prompt.value) {
      router.push('/404')
    }
  }, 5000)
```

File: src/components/Posts/Comments/DisplayComment.vue (hunk 1 above)
Comment (arnonrdp) [reply src/components/Posts/Comments/DisplayComment.vue:151]: Ok, let's discuss tomorrow in the meeting

File: src/pages/PromptPage.vue (hunk 4 above)
Comment (arnonrdp) [reply src/pages/PromptPage.vue:78]: Noted, let's test it in `develop`.

**Review by arnonrdp — APPROVED** [review]

**Pull request merged**